### ISSUE TITLE
Rescale values automatically before and after a k-space augmentation

### ISSUE THREAD

**Issue by fepegar** [fepegar/torchio#95]
Currently, a warning is issued if the image contains negative values, as they will be lost after keeping only the absolute value from the result of an inverse Fourier transform.

A solution to avoid this would be to rescale the data to positive values only, apply the transform, and rescale back to the original range. This would assume that the transform has not modified the min and max too much.

**Comment by romainVala** [comment]
This may becomes very handy.
min max will change ...

I would say,  it is the user responsability to compose correctly the transform so that it make sense ...

**Comment by fepegar** [comment]
I don't think you meant to say "handy" :)

The scale would be preserved if the original range is stored. What I propose is something like this:

```python
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
sns.set()

np.random.seed(0)

sample = np.random.randn(10000)*50 + 100

original_min, original_max = sample.min(), sample.max()
original_range = original_max - original_min
rescaled = (sample - original_min) / original_range  # [0, 1]

rescaled += (np.random.rand(10000) - 0.5) * 0.01

transformed = rescaled * original_range
transformed += original_min

sns.distplot(sample, hist=False)
sns.distplot(transformed, hist=False)
plt.show()
```

The example is not great, but it shows how the range is generally preserved between the input and the output.

**Comment by fepegar** [comment]
Fixed in #207.
